homeassistant/helpers/template.py

```diff
@@ -1314,12 +1319,59 @@ def urlencode(value):
     return urllib_urlencode(value).encode("utf-8")
 
 
+def _render_with_context(
+    template_str: str, template: jinja2.Template, **kwargs: Any
```

Comment:
It took me a bit to realize, this is wrapping Jinja2 objects. Why don't we store our own template objects in it so we can access the template string too? 

In the future we can then add "names" to templates, ie `sensor.kitchen_temp - attribute name`